src/GafferArnold/InteractiveArnoldRender.cpp

```diff
@@ -91,6 +91,9 @@ void InteractiveArnoldRender::flushCaches( int flags )
 
 		if( !instance->renderer() )
 		{
+			// No renderer found so flush global cache so that new renders spinning up later will see a refereshed
+			// texture cache. This is needed because apparently Arnold only has one texture cache between universes.
+			AiUniverseCacheFlush( nullptr, flags );
```

Comment:
I think maybe there's another failure case we're not handling. If there aren't any InteractiveArnoldRender nodes at all, then we won't get here. So maybe this needs to be moved after the loop, and run only if we didn't flush anything in the loop?

Is this fix need only for textures or also the other cache types? Should we be checking `flags` to see if we need to do it at all?

Reply:
Depends if it makes sense for the user to ask to flush the cache if there are no interactive render nodes.

I need to double check for the other cache types as I don't remember of the top of my head. 

I'll double check what the Arnold folks said and then update the PR.

Unit tests sounds like a great idea - I'll get on it :)